tools-local/unix/invoke-extension.sh

```diff
@@ -0,0 +1,116 @@
+    echo "  repository-root: Path to repository root."
+    echo "  tool-name: Name of the tool."
+    echo "  override-scripts-folder-path: If a path is specified then, search and acquire scripts from the specified folder will be invoked."
+    echo "                                  Otherwise, search will use the default search and acquire scripts located within the repository."
+    echo "  strict-tool-version-match: If equals to \"strict\" then, search will ensure that the version of the tool searched is the declared version."
+    echo "                                          Otherwise, search will attempt to find a version of the tool, which may not be the declared version."
+    echo "  tool-path: Path to the tool executable."
+    echo ""
+    echo "Checks if the specified tool has its own implementation of the search or acquire script. If so, invokes the corresponding script. Otherwise, invokes the base implementation."
+    echo ""
+    echo "Example #1"
+    echo "invoke-extension.sh search-tool.sh \"/Users/dotnet/corefx\" cmake """
+    echo "  Searches for CMake, not necessarily the declared version, using the default search scripts located within the repository."
+    echo ""
+    echo "Example #2"
+    echo "invoke-extension.sh acquire-tool.sh \"/Users/dotnet/corefx\" cmake """
+    echo "  Acquires the declared version of CMake, using the default search scripts located within the repository."
+    echo ""
+    echo "Example #3"
+    echo "invoke-extension.sh search-tool.sh \"/Users/dotnet/corefx\" cmake \"/Users/dotnet/MyCustomScripts\" strict"
+    echo "  Searches for the declared version of CMake using the search scripts located in \"/Users/dotnet/MyCustomScripts\"."
+    echo ""
+    echo "Example #4"
+    echo "invoke-extension.sh get-version.sh \"/Users/dotnet/corefx\" cmake "" "" \"/Users/dotnet/corefx/Tools/download/cmake/bin/cmake\" "
+    echo "  Get the version number of CMake executable located at /Users/dotnet/corefx/Tools/download/cmake/bin/cmake\"."
+    echo ""
+}
+
+if [ $# -lt 4 ]; then
+    usage
+    exit 1
+fi
+
+extensionScriptName="$1"
+shift
+
+repoRoot="$1"
+toolName="$2"
+overrideScriptsFolderPath="$3"
+strictToolVersionMatch="$4"
+toolPath="$5"
+
+if [ -z "$extensionScriptName" ]; then
+    echo "Argument passed as script-name is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+if [ -z "$repoRoot" ]; then
+    echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+if [ -z "$toolName" ]; then
+    echo "Argument passed as tool-name is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+if [ ! -z "$overrideScriptsFolderPath" ] &&  [ ! -d "$overrideScriptsFolderPath" ]; then
+    echo "Path specified as override-scripts-folder-path does not exist or is not accessible. Path: $overrideScriptsFolderPath"
+    usage
+    exit 1
+fi
+
+if [ ! -z "$toolPath" ] && [ ! -f "$toolPath" ]; then
```

Comment:
If a path was specified then check if the file exists. Otherwise, skip the check. This is unlike the earlier PRs where I was checking for empty and exists.